boards/sim/sim/sim/src/sim_bringup.c

```diff
@@ -311,5 +316,15 @@ int sim_bringup(void)
     }
 #endif
 
+#ifdef CONFIG_SIM_LINUXI2CBUS
```

Comment:
all code outside i2c shouldn't contain LINUX/linux term, these words should hide inside the implementation detail.

Reply:
What do you think this function should look like then if it is not Linux bus specific? In Linux I need to pass something like
`/dev/i2c-#` and, but for macOS or Windows we would instead have to interacts via a non i2c interface that the controller itself is on.  For the MCP2221 that I happen to be using that would be via USB HID which would take something like VID, PID

Reply:
The prototype could keep as before, but linux need remove:
```
#ifdef CONFIG_SIM_I2CBUS
up_i2c_initialize(xxx);
#endif
```
up_i2c_initialize for different host can provide the different implementation to map the busid to the host specific identifer. The key is that the function name need be generic, so the caller don't care how the interface get implemented, but the prototype need be enough flexible to give the host freedom.